docs/_static/extra_css.css

```diff
@@ -16,11 +16,51 @@
   --md-icon__fa-bitbucket: url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path d="M22.2 32A16 16 0 0 0 6 47.8a26.35 26.35 0 0 0 .2 2.8l67.9 412.1a21.77 21.77 0 0 0 21.3 18.2h325.7a16 16 0 0 0 16-13.4L505 50.7a16 16 0 0 0-13.2-18.3 24.58 24.58 0 0 0-2.8-.2L22.2 32zm285.9 297.8h-104l-28.1-147h157.3l-25.2 147z"/></svg>');
-  --md-admonition-icon--pied-piper: url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 576 512"><path d="M244 246c-3.2-2-6.3-2.9-10.1-2.9-6.6 0-12.6 3.2-19.3 3.7l1.7 4.9zm135.9 197.9c-19 0-64.1 9.5-79.9 19.8l6.9 45.1c35.7 6.1 70.1 3.6 106-9.8-4.8-10-23.5-55.1-33-55.1zM340.8 177c6.6 2.8 11.5 9.2 22.7 22.1 2-1.4 7.5-5.2 7.5-8.6 0-4.9-11.8-13.2-13.2-23 11.2-5.7 25.2-6 37.6-8.9 68.1-16.4 116.3-52.9 146.8-116.7C548.3 29.3 554 16.1 554.6 2l-2 2.6c-28.4 50-33 63.2-81.3 100-31.9 24.4-69.2 40.2-106.6 54.6l-6.3-.3v-21.8c-19.6 1.6-19.7-14.6-31.6-23-18.7 20.6-31.6 40.8-58.9 51.1-12.7 4.8-19.6 10-25.9 21.8 34.9-16.4 91.2-13.5 98.8-10zM555.5 0l-.6 1.1-.3.9.6-.6zm-59.2 382.1c-33.9-56.9-75.3-118.4-150-115.5l-.3-6c-1.1-13.5 32.8 3.2 35.1-31l-14.4 7.2c-19.8-45.7-8.6-54.3-65.5-54.3-14.7 0-26.7 1.7-41.4 4.6 2.9 18.6 2.2 36.7-10.9 50.3l19.5 5.5c-1.7 3.2-2.9 6.3-2.9 9.8 0 21 42.8 2.9 42.8 33.6 0 18.4-36.8 60.1-54.9 60.1-8 0-53.7-50-53.4-60.1l.3-4.6 52.3-11.5c13-2.6 12.3-22.7-2.9-22.7-3.7 0-43.1 9.2-49.4 10.6-2-5.2-7.5-14.1-13.8-14.1-3.2 0-6.3 3.2-9.5 4-9.2 2.6-31 2.9-21.5 20.1L15.9 298.5c-5.5 1.1-8.9 6.3-8.9 11.8 0 6 5.5 10.9 11.5 10.9 8 0 131.3-28.4 147.4-32.2 2.6 3.2 4.6 6.3 7.8 8.6 20.1 14.4 59.8 85.9 76.4 85.9 24.1 0 58-22.4 71.3-41.9 3.2-4.3 6.9-7.5 12.4-6.9.6 13.8-31.6 34.2-33 43.7-1.4 10.2-1 35.2-.3 41.1 26.7 8.1 52-3.6 77.9-2.9 4.3-21 10.6-41.9 9.8-63.5l-.3-9.5c-1.4-34.2-10.9-38.5-34.8-58.6-1.1-1.1-2.6-2.6-3.7-4 2.2-1.4 1.1-1 4.6-1.7 88.5 0 56.3 183.6 111.5 229.9 33.1-15 72.5-27.9 103.5-47.2-29-25.6-52.6-45.7-72.7-79.9zm-196.2 46.1v27.2l11.8-3.4-2.9-23.8zm-68.7-150.4l24.1 61.2 21-13.8-31.3-50.9zm84.4 154.9l2 12.4c9-1.5 58.4-6.6 58.4-14.1 0-1.4-.6-3.2-.9-4.6-26.8 0-36.9 3.8-59.5 6.3z"/></svg>')
+}
+
+/* ************************* my-special-key style ************************************* */
+
+.md-typeset kbd.key-my-special-key,
+.md-typeset kbd.key-git {
+  color: var(--md-primary-fg-color);
+  background: var(--md-accent-bg-color);
+}
+
+.md-typeset kbd.key-git::before {
+  content: "██";
+  -webkit-mask-image: var(--md-icon__fa-git);
+  -webkit-mask-repeat: no-repeat;
+  mask-image: var(--md-icon__fa-git);
+  mask-repeat: no-repeat;
+}
+
+.md-typeset kbd.key-my-special-key:hover,
+.md-typeset kbd.key-git:hover {
+  color: var(--md-accent-fg-color);
+}
+
+/* **************************** custom-task-list style rules *************************** */
+
+/* style for the unchecked checkboxes */
+.custom-task-list-style>ul>.task-list-item>.task-list-control>.task-list-indicator::before {
```

Comment:
>  For example we might add a way to customize the styling from conf.py that doesn't require copying any SVG files, etc.

I'd feel more confident in this idea after we have all the SVG data exported in an optional CSS file. For now, it doesn't get any simpler than using the `:class:` option and adding some extra CSS.

Reply:
I think we can avoid having to export all of the SVG data into CSS (as that would waste space by including SVGs that are not actually needed).  Instead as part of the sphinx build we could collect all of the SVGs that are referenced, and output them to a new CSS file.

Reply:
(Currently all of the CSS for our theme is static, but it doesn't have to be that way.)

Reply:
> Instead as part of the sphinx build we could collect all of the SVGs that are referenced

How do you imagine doing this? I'm willing to hack this together, but I want to share your vision about implementation (how the user would go about specifying which SVGs are used). Initially, I imagine a list (in conf.py - probably part of `html_theme_options`) of filenames (with path relative to .icons folder).

In upstream, squidfunk uses the `pymdownx.emoji` ext to specify inline SVGs (as exemplified in [icons & emojis page](https://squidfunk.github.io/mkdocs-material/reference/icons-emojis/#using-icons)). That approach is very similar to what I submitted to sphinx-design (for inline icon roles - still waiting on that release though 😞 ).

Reply:
I imagined that we might have various Python functions or `conf.py` options that control customization of icons --- e.g. specifying custom admonition types with custom icons, or custom task list styles, since the actual CSS required to define those customizations ends up "leaking" a lot of implementation details of the theme, so it would be best if the user didn't need to know about that.

Then there might be a lower level API for registering specific icons that need to be added to the CSS.  The implementation of those higher-level customization options would make use of this lower-level API to ensure the icons they need are available.  The user could also directly make use of this lower level API / `conf.py` option if they have written custom CSS that requires certain icons as well.

Reply:
As for the custom admonition icons, I can add an option (call it "custom-icon") to the md-admonition directive that lets users specify the relative path to the SVG within the .icons folder. 

Thinking about it further, I suppose the task-list directive could use the same treatment, probably using the directive's `:custom:` option or the `custom_checkboxes` conf.py option (which are both currently boolean values).

Outside of special cases like that I think I'd rather defer to using the sphinx-design ext to use inline icons in text.

I noticed that [upstream relies on page metadata](https://squidfunk.github.io/mkdocs-material/reference/#setting-the-page-icon) to give the page title a prefixed icon. It also recommends altering the [color of icons with a user-specified CSS](https://squidfunk.github.io/mkdocs-material/reference/icons-emojis/#with-colors) class (& corresponding CSS rules), so eventually the user may have to understand fundamental CSS at some point.

---
BTW, I did simplify this CSS selector by using the elements' tag names instead of using primarily ext-specific CSS classes:
```css
.custom-task-list-style > ul > li > label .task-list-indicator::before {
  /* ... */
}
```

Reply:
I don't think it makes sense to worry about it as part of this pull request, but yeah, adding an option to `md-admonition` and other directives seems like a good idea.

As I see it, in most cases users wouldn't directly specify such an option within their rST sources.  Instead, they would define their own custom admonition directives by inheriting from the class that defines `md-admonition`, and would specify the icon there.

E.g. they could define an `example` admonition or something and specify a custom icon to use for it.  That way they can maintain a separation between the semantic content of their documents and any style information.

I think it is not so bad for users to have to specify CSS as long as they can specify their styles in a simple way that doesn't depend on implementation details of the theme.  E.g. in the example you posted (https://squidfunk.github.io/mkdocs-material/reference/icons-emojis/#with-colors) they are just defining a CSS class and specifying a `color` --- nothing complicated about that.

Reply:
ok, resolving this thread since the selectors have been simplified. I'm glad to have started discussing how to implement user specified icons though.